src/main/scala/de/heikoseeberger/sbtheader/License.scala

```diff
@@ -16,25 +16,13 @@
 
 package de.heikoseeberger.sbtheader
 
-import de.heikoseeberger.sbtheader.CommentStyle.CStyleBlockComment
-
-import scala.util.matching.Regex
-
 sealed trait License {
-
-  def apply(yyyy: String,
-            copyrightOwner: String,
-            commentStyle: CommentStyle = CStyleBlockComment): (Regex, String) = {
-    val text = createLicenseText(yyyy, copyrightOwner)
-    commentStyle(text)
-  }
-
-  def createLicenseText(yyyy: String, copyrightOwner: String): String
+  def text: String
 }
 
 trait Licenses {
-  object Apache2_0 extends License {
-    override def createLicenseText(yyyy: String, copyrightOwner: String) =
+  case class Apache2_0(yyyy: String, copyrightOwner: String) extends License {
```

Comment:
@hseeberger I just realized, that compiling the code now issues warnings:

```
[warn] /Users/bene/workspace/sbt/sbt-header/src/main/scala/de/heikoseeberger/sbtheader/License.scala:24: The outer reference in this type test cannot be checked at run time.
[warn]   final case class Apache2_0(yyyy: String, copyrightOwner: String) extends License {
```

Do you think this is a problem?